Move `Prompt` object instantiation from server to prompt manager

ramjibc opened the pull request:
The `Prompt()` decorator in `server.py` is instantiating a Prompt object before adding it to the list of prompts. This is inconsistent with how the `Tool()` decorator works; it delegates tool instantiation to the tool manager. 

## Motivation and Context
To make the decorators consistent and create clear separation of concerns: I have moved prompt object instantiation from `server.py` to `prompts/manager.py`.

## How Has This Been Tested?
I updated and ran all unit tests.

## Breaking Changes
No. Users will not be impacted as the `prompt()` decorator's signature remains unchanged.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist
- [X] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [X] My code follows the repository's style guidelines
- [X] New and existing tests pass locally
- [X] I have added appropriate error handling
- [ ] I have added or updated documentation as needed

## Additional context
Separation of concerns: instantiation and managements of prompt objects should be done by the prompt manager, not the server.

ihrpr reviewed (changes requested):
Thank you for working on this!
There are a few concerns:

1. API Change for Direct Usage: The add_prompt() method signature changes from accepting a Prompt to accepting a function. This breaks backwards compatibility.
 2. Less Flexibility: With the old design, users could create custom Prompt objects with specific configurations before
  adding them. Now they're forced to use the manager's instantiation logic.
 3. Inconsistent Manager APIs: While this makes server decorators consistent, it makes the manager APIs inconsistent:
    - ToolManager.add_tool() accepts both functions and Tool objects
    - PromptManager.add_prompt() now only accepts functions

ramjibc commented:
Thank you for the review @ihrpr . I have addressed your comments, but primarily from the backward compatibility perspective. Also, I don't see this in code

> Inconsistent Manager APIs: While this makes server decorators consistent, it makes the manager APIs inconsistent:
ToolManager.add_tool() accepts both functions and Tool objects
PromptManager.add_prompt() now only accepts functions

I see `ToolManager.add_tool()` taking only a function as input.

In any case, I have made changes such that the code doesn't break for existing users but moves it a bit closer to achieving clear separation of concerns. Let me know your comments.
